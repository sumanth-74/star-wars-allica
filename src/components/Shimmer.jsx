import React from 'react';
import '../App.css'; // Ensure this import is correct

const Shimmer = ({ count = 12 }) => {
  const shimmerItems = Array.from({ length: count });

  return (
    <div className="shimmer-container">
      {shimmerItems.map((_, index) => (
        <div key={index} className="shimmer-item"></div>
      ))}
    </div>
  );
};

export default Shimmer;
