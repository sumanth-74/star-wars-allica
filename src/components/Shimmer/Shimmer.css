
.shimmer-container {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  gap: 1rem;
  margin-top: 1rem;
}

.shimmer-item {
  background: linear-gradient(90deg, #f0f0f0 25%, #e0e0e0 50%, #f0f0f0 75%);
  background-size: 200% 100%;
  animation: shimmer 1.5s infinite;
  height: 150px;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  width: 100%;
}

@keyframes shimmer {
  0% {
    background-position: -200% 0;
  }
  100% {
    background-position: 200% 0;
  }
}
