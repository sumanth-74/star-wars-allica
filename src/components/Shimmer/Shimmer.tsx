import React from 'react';
import './Shimmer.css';

const Shimmer = ({ count = 12 }) => {
  const shimmerItems = Array.from({ length: count });

  return (
    <div className="shimmer-container">
      {shimmerItems.map((_, index) => (
        <div key={index} className="shimmer-item"></div>
      ))}
    </div>
  );
};

export default Shimmer;
