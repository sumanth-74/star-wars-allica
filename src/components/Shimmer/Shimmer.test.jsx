import React from 'react';
import { render } from '@testing-library/react';
import Shimmer from './Shimmer';

describe('Shimmer Component', () => {
  it('renders without crashing', () => {
    const { container } = render(<Shimmer />);
    expect(container.firstChild).toHaveClass('shimmer-container');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Shimmer />);
    expect(asFragment()).toMatchSnapshot();
  });
});