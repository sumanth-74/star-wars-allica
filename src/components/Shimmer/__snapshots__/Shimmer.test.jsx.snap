// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shimmer Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="shimmer-container"
  >
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
    <div
      class="shimmer-item"
    />
  </div>
</DocumentFragment>
`;
